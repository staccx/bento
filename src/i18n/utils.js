import SanityBlockContent from "../sanity/components/SanityBlockContent"
import Text from "../components/Text/Text/Text"
import React from "react"

export const isBlock = item =>
  item &&
  item.hasOwnProperty("_type") &&
  (item._type === "block" || item._type === "image")

export const isRichText = val =>
  Array.isArray(val) &&
  val.some(
    arr =>
      typeof arr !== "string" &&
      Array.isArray(arr) &&
      arr.some(item => item._type)
  )

export const getComponent = val => {
  return Array.isArray(val) ? (
    val.map(getComponent)
  ) : isBlock(val) ? (
    <SanityBlockContent key={val._key} blocks={val} />
  ) : (
    <Text key={val}>{val}</Text>
  )
}

export const handleArray = (value, data, children, translate) => {
  const values = value.map(
    (k, index) => translate(k, data) || React.Children(children)[index]
  )

  if (typeof children === "function") {
    return children(values)
  }

  return values.map(getComponent)
}
