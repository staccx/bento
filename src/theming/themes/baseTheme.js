import React from "react"
import Theme from "../Theme"
import reset from "../styles/cssResets"
import grids from "../../components/Layout/Layout/Layout.Grids"
import codeTheme from "../codeTheme"

const Logo = () => <div>ThemeComponent for BaseTheme</div>
Logo.displayName = "Logo for base theme"

export const theme = {
  name: "Base Theme",
  storybookName: "Base Theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190", // for toggle eks
    link: "#279AF1"
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px",
    lineHeight: "1.5"
  },
  fontWeight: {
    bold: "bold",
    normal: "normal",
    light: "lighter"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px",
  reset,
  grid: grids,
  logo: Logo
}

export const StyleguideTheme = new Theme(theme, codeTheme)

export default theme
