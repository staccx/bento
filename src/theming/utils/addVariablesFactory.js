import set from "lodash.set"
import { Theme, themes } from "../index"
import { themeLogger } from "../theme.logger"
import { getDisplayName } from "./getDisplayName"

export const addVariablesFactory = component => (
  variables,
  themeName = "_default"
) => {
  if (!themes.hasOwnProperty(themeName)) {
    themeLogger.warn(
      `Theme with name ${themeName} does not exists. One will be created`
    )
    themes[themeName] = new Theme()
  }
  const theme = themes[themeName]

  if (!component) {
    throw new Error("No component")
  }

  if (!variables) {
    throw new Error("addVariables called with no variables")
  }

  for (const key of Object.keys(variables)) {
    const variant = variables[key]

    const styles = {}
    set(styles, `variables.${getDisplayName(component)}.${key}`, variant)
    theme.append(styles)
  }
}
