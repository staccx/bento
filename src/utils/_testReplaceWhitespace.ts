export const replaceWhiteSpaceForMatcher = value =>
  value ? value.replace(/\s/g, "\u00a0") : value
