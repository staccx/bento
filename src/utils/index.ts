export { stringIncludes } from "./stringIncludes"
export { distinct } from "./distinct"
export { flatten } from "./flatten"
export { partition } from "./partition/partition"
