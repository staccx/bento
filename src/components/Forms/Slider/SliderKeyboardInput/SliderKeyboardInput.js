import React, { useState } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import Slider, { SliderProps } from "../Slider/Slider"
import {
  applyVariants,
  targetSize,
  fontFamily,
  font,
  color
} from "../../../../theming"
import themeProps from "./SliderKeyboardInput.themeProps"
import { componentCreateFactory } from "../../../../theming/utils/createVariantsFunctionFactory"
import CurrencyInput, {
  CurrencyInputProps
} from "../../Input/CurrencyInput/CurrencyInput"

/**
 * Slider component which can be defined by various properties, i.e. min, max, step, and can be controlled by keyboard inputs
 */
const SliderKeyboardInput = ({
  label,
  name,
  min,
  max,
  step,
  mask,
  className,
  variant,
  onChange,
  showLabel,
  defaultValue,
  inputProps,
  sliderProps,
  ...restProps
}) => {
  const [inputValue, setInputValue] = useState(defaultValue)
  const [sliderValue, setSliderValue] = useState(defaultValue)
  const [stepValue, setStepValue] = useState(step)

  const handleInputFocus = () => {
    setStepValue(1)
  }

  const handleInputChange = value => {
    if (!value) {
      setInputValue("")
    } else {
      setInputValue(value)

      if (parseInt(value, 10) >= min && parseInt(value, 10) <= max) {
        setSliderValue(value)
      }

      if (parseInt(value, 10) >= max) {
        setSliderValue(max)
      }

      if (parseInt(value, 10) <= min) {
        setSliderValue(min)
      }

      onChange && onChange(value)
    }
  }

  const handleSliderChange = value => {
    setInputValue(parseInt(value, 10))
    setSliderValue(value)
    onChange && onChange(value)
  }

  const handleSliderStart = () => {
    setStepValue(step)
  }

  return (
    <SliderWrapper className={className} variant={variant} {...restProps}>
      <SliderInput
        type="text"
        name={`${name}-keyboard`}
        id={`${name}-keyboard`}
        value={inputValue}
        onChange={e => handleInputChange(e.target.rawValue)}
        onFocus={handleInputFocus}
        // onBlur={handleBlur}
        label={showLabel ? label : null}
        min={min}
        max={max}
        mask={mask || null}
        // disabled={isAnimatingIn}
        variant={variant}
      />
      <Slider
        value={sliderValue}
        name={name}
        id={name}
        step={stepValue}
        min={min}
        max={max}
        onChange={handleSliderChange}
        onSlideStart={handleSliderStart}
        onSlideEnd={handleSliderChange}
        defaultValue={defaultValue}
        variant={variant}
      />
    </SliderWrapper>
  )
}

const SliderWrapper = styled.div`
  font-family: ${fontFamily.body};
  font-size: ${font.input};
  ${applyVariants(themeProps.wrapper)};
`

const SliderInput = styled(CurrencyInput)`
  background-color: transparent;
  border: 0;
  min-height: ${targetSize.normal};
  max-width: 100%;
  -moz-appearance: textfield;
  outline: none;

  &::-webkit-inner-spin-button,
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  &:hover,
  &:active,
  &:focus,
  &:-moz-ui-invalid,
  &:invalid {
    outline: none;
    box-shadow: none;
  }

  &:disabled {
    color: ${color.text};
  }
  ${applyVariants(themeProps.input)};
`

SliderKeyboardInput.defaultProps = {
  className: "",
  mask: null,
  max: 100,
  min: 0,
  onChange: null,
  step: 1,
  value: 0,
  showLabel: true
}

SliderKeyboardInput.propTypes = {
  className: PropTypes.string,
  easingFunction: PropTypes.func,
  label: PropTypes.string.isRequired,
  mask: PropTypes.func,
  max: PropTypes.number,
  min: PropTypes.number,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  step: PropTypes.number,
  variant: PropTypes.string,
  value: PropTypes.number,
  showLabel: PropTypes.bool,
  inputProps: CurrencyInputProps,
  sliderProps: PropTypes.shape(SliderProps)
}
SliderKeyboardInput.themeProps = themeProps
SliderKeyboardInput.createVariants = componentCreateFactory(SliderKeyboardInput)

export default SliderKeyboardInput
