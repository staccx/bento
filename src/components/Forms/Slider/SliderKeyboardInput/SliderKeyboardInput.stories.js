import React from "react"
import SliderKeyboardInput from "./SliderKeyboardInput"

export default {
  title: "Components/Forms/Slider/SliderKeyboardInput",
  component: SliderKeyboardInput
}

export const Default = args => <SliderKeyboardInput {...args} />
Default.args = {
  label: "Slider with keyboard input",
  min: 100,
  max: 500,
  step: 50,
  defaultValue: 250,
  onChange: value => console.log("The value in default is ", value)
}

export const DefaultValueOutsideOfStepIncrement = args => (
  <SliderKeyboardInput {...args} />
)
DefaultValueOutsideOfStepIncrement.args = {
  name: "stepoutside",
  label: "Slider with keyboard input",
  min: 100,
  max: 500,
  step: 50,
  defaultValue: 251,
  onChange: value => console.log("The value step e is ", value)
}
