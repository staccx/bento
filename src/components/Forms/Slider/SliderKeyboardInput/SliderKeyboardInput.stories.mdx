import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import SliderKeyboardInput from "./SliderKeyboardInput"

<Meta title="Components|Forms/SliderKeyboardInput" component={SliderKeyboardInput}/>


# Slider


```javascript
import {SliderKeyboardInput} from "@staccx/bento"
```

## Example usage
<Preview>
  <Story name="Default">
    <SliderKeyboardInput label="Slider with keyboard input" min={100} max={500} step={50} defaultValue={250}  />
  </Story>
</Preview>

<Preview>
  <Story name="Default value outside of step increment">
    <SliderKeyboardInput name="stepoutside" label="Default value outside of step increment" min={100} max={500} step={50} defaultValue={251}  />
  </Story>
</Preview>

<Preview>
  <Story name="Big un">
    <SliderKeyboardInput name="big" label="Large range, low step" min={10} max={100000} step={10} defaultValue={65356}  />
  </Story>
</Preview>

## Props for SliderKeyboardInput
<Props of={SliderKeyboardInput} />
