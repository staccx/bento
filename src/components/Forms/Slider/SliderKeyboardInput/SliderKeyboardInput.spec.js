import React from "react"
import { render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import SliderKeyboardInput from "./SliderKeyboardInput"
import { replaceWhiteSpaceForMatcher } from "../../../../utils/_testReplaceWhitespace"

const setup = (props = {}) => {
  const utils = render(
    <ThemeProvider theme={baseTheme}>
      <SliderKeyboardInput
        name="slider"
        label="slider"
        min={0}
        max={200}
        step={1}
        {...props}
      />
    </ThemeProvider>
  )
  const sliderKeyboardLabel = utils.getByLabelText("slider")
  return {
    sliderKeyboardLabel,
    ...utils
  }
}

describe("SliderKeyboardInput", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Name", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput name="slider" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("showLabel", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput showLabel />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("value", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput value={100} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("step", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput step={1} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("onBlur", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput onBlur={() => console.log("test")} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("onChange", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput onChange={() => console.log("test")} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("min", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput min={0} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("max", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput max={1000} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("label", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput label="Label" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("all", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput
              showLabel
              step={1}
              value={100}
              name="slider"
              label="Label"
              onBlur={() => console.log("test")}
              onChange={console.log("virker")}
              max={1000}
              min={0}
            />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("simulated keyboard with no value", () => {
      const { sliderKeyboardLabel } = setup()
      expect(sliderKeyboardLabel.value).toBe("")
    })
    it("simulated keyboard with default value", () => {
      const { sliderKeyboardLabel } = setup({ defaultValue: 1337 })
      expect(sliderKeyboardLabel.value).toBe(
        replaceWhiteSpaceForMatcher("1,337")
      )
    })
  })
})

describe("Rendering", () => {
  it("No props", () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <SliderKeyboardInput showLabel name="slider" label="Label" />
      </ThemeProvider>
    )
    expect(screen.getByLabelText("Label")).toBeInTheDocument()
  })
})
