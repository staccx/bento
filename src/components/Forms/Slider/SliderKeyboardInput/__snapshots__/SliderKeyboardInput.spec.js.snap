// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderKeyboardInput Snapshots Name 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="slider-keyboard"
      mask={null}
      max={100}
      min={0}
      name="slider-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={NaN}
          className="sc-bdfBwQ fHavKc sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz bJfODj"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": NaN,
              "value": NaN,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots No props 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={100}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots all 1`] = `
<div
  className="sc-bqyKva hYthuH"
  onBlur={[Function]}
  value={100}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <label
      className="sc-iBPRYJ hrNrxW sc-iqHYGH gHOQQy"
      htmlFor="slider-keyboard"
    >
      Label
    </label>
    <input
      className="sc-dQppl jYXCnb"
      id="slider-keyboard"
      mask={null}
      max={1000}
      min={0}
      name="slider-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={1000}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots label 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <label
      className="sc-iBPRYJ hrNrxW sc-iqHYGH gHOQQy"
      htmlFor="sliderKeyboard-keyboard"
    >
      Label
    </label>
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={100}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots max 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={1000}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={1000}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots min 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={100}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots onBlur 1`] = `
<div
  className="sc-bqyKva hYthuH"
  onBlur={[Function]}
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={100}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots onChange 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={100}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots showLabel 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={100}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots step 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={0}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={100}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots value 1`] = `
<div
  className="sc-bqyKva hYthuH"
  value={100}
>
  <div
    className="sc-crrsfI leevFK sc-kstrdz hZYYUJ"
  >
    <input
      className="sc-dQppl jYXCnb"
      id="sliderKeyboard-keyboard"
      mask={null}
      max={100}
      min={0}
      name="sliderKeyboard-keyboard"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-dlfnbm jQQXla"
  >
    <div
      className="sc-hKgILt cNgtla"
      style={Object {}}
    >
      <div
        className="sc-eCssSg fXBLJQ"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-bdfBwQ dMvusR sc-jSgupP fbijvz"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-gKsewC eoWRga"
      >
        <div
          className="sc-gsTCUz XesgK"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
